CMakeLists.txt

```diff
@@ -791,7 +791,7 @@ hpx_check_for_cxx11_std_unique_ptr(
   REQUIRED "HPX needs support for C++11 std::unique_ptr")
 
 hpx_check_for_cxx11_std_type_traits(
-  DEFINITIONS HPX_HAVE_CXX11_STD_TYPE_TRAITS)
+  REQUIRED "HPX needs support for C++11 basic type traits")
```

Comment:
This makes the intel builders that use a gcc 4.4. stdlib fail. I am not sure if dropping support for those is a good idea right now.